ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -38,22 +37,19 @@ bool joystick_exists(int id) {
 }
 
 std::string joystick_name(int id) {
-  const char *name;
-  name = SDL_JoystickName(joystick_get_handle(id));
+  const char *name = SDL_JoystickName(joystick_get_handle(id));
   return (name ? name : "");
 }
 
 int joystick_axes(int id) {
-  int axes;
   SDL_Joystick *joystick = joystick_get_handle(id);
-  axes = SDL_JoystickNumAxes(joystick);
+  int axes = SDL_JoystickNumAxes(joystick);
   return (axes > 0 ? axes : 0);
 }
 
 int joystick_buttons(int id) {
-  int buttons;
   SDL_Joystick *joystick = joystick_get_handle(id);
-  buttons = SDL_JoystickNumButtons(joystick);
+  int buttons = SDL_JoystickNumButtons(joystick);
   return (buttons > 0 ? buttons : 0);
```

Comment:
If a negative value is returned it means something has gone wrong. I think we should report this in debug mode.
```cpp
#ifdef DEBUG_MODE
if (buttons < 0) DEBUG_MESSAGE("Failed to query jostick id  + SDL_GetError());
#endif
```

Reply:
You mean opening and closing based on this? https://github.com/enigma-dev/enigma-dev/blob/5881bfb3a6eb53b67d30f2b7ae6fd72211a69ca1/ENIGMAsystem/SHELL/Platforms/SDL/Gamepad.cpp#L55